First commit

borodatyidrug opened the pull request:
Made everything.

borodatyidrug committed:
First commit

Made everything.

Lada303 commented on src/main/java/com/edu/ulab/app/entity/Book.java:
```diff
@@ -0,0 +1,16 @@
+
+@Data
+@Builder
+public class Book implements Idable<Long> {
```
тут еще можно несколько аннотаций накинуть

Lada303 commented on src/main/java/com/edu/ulab/app/exception/NotFoundException.java:
```diff
@@ -0,0 +1,7 @@
+    public NotFoundException(String message) {
+        super(message);
+    }
+}
```
добавь свое кастомное исключение и опиши его обработку в Хэндлере

Lada303 commented on src/main/java/com/edu/ulab/app/facade/UserDataFacade.java:
```diff
@@ -0,0 +1,138 @@
+    }
+
+    public void deleteUserWithBooks(Long userId) {
+    	userService.getUserById(userId).getBooksId().stream().forEach(bookService::deleteBookById);
```
а тут лог?

Lada303 commented on src/main/java/com/edu/ulab/app/service/UserService.java:
```diff
@@ -0,0 +1,11 @@
+
+public interface UserService {
+    UserDto createUser(UserDto userDto);
+    UserDto updateUser(UserDto userDto);
```
обычно все же пропускают сроки между методами

Lada303 replied:
и пишут джавадоки на них

Lada303 commented on src/main/java/com/edu/ulab/app/storage/Storage.java:
```diff
@@ -0,0 +1,114 @@
+			log.debug("Saved new object with id {}", generatedId);
+			return generatedId;
+		} else {
+			throw new IllegalArgumentException("Only non null values are allowed");
```
вот эти проверки делаются или в сервисном слое или в фасаде -  и там уже кидают исключения. В слое работы с бд - только запрос-ответ. все

borodatyidrug committed:
Внесены исправления по замечаниям преподавателя, а так же устранены
некоторые незначительные недочеты.

Добавлено логгирование в метод deleteUserWithBooks() в классе
UserDataFacade, добавлены javadoc и перенесены проверки входных данных
со слоя хранилища на слой сервисов.

Lada303 commented on src/main/java/com/edu/ulab/app/service/impl/BookServiceImpl.java:
```diff
@@ -0,0 +1,88 @@
+    @Override
+    public BookDto getBookById(Long id) {
+    	if (id != null) {
+    		Book book = (Book) storage.get(id);
```
делают не так if (id == null) {throw} , а далее просто пишут код работы  - так более читабельно

Lada303 commented on src/main/java/com/edu/ulab/app/web/handler/ControllerExceptionHandler.java:
```diff
@@ -0,0 +1,46 @@
+    @ExceptionHandler(NotFoundException.class)
+    public ResponseEntity<BaseWebResponse> handleNotFoundExceptionException(@NonNull final NotFoundException exc) {
+        log.error(exc.getMessage());
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST)
```
тут и статус Not_Found
